internal/provider/resource_k8s.go

```diff
@@ -989,6 +989,9 @@ func (k *K8sDeployer) getK8sFreeIP(ipRange gridtypes.IPNet, nodeID uint32) (stri
 		if !isInByte(k.NodeUsedIPs[nodeID], i) {
 			k.NodeUsedIPs[nodeID] = append(k.NodeUsedIPs[nodeID], i)
 			ip := ipRange.IP.To4()
+			if ip == nil {
```

Comment:
if you want to convert to byte, do it inside of the loop not in the initializer and the condition, The check on the ip should happen very early in the function, before the slice append or maybe even outside of the isInByte check or the loop even. The whole function seems to be confusing ( I assume you're working against the last segment of an ip of /24 cidr? also if that's the case the check against the last segment, it should be called hostID )